src/datasets/vtr_dataset.py

```diff
@@ -17,27 +19,38 @@ def __init__(
         window_size: int = 30,
         stride: int = 5,
         max_seq_len: int = 512,
+        ratio: float = 0.7,
+        ocr_flag: bool = False,
     ):
         logger.info(f"Initializing VTRDataset with {len(labeled_texts)} samples, use max seq len {max_seq_len}")
         self.labeled_texts = labeled_texts
         self.max_seq_len = max_seq_len
+        self.ocr_flag = ocr_flag
 
+        self.slicer_ocr = VTRSlicerOCR(
+            font=font, font_size=font_size, window_size=window_size, stride=stride, ratio=ratio
+        )
         self.slicer = VTRSlicer(font=font, font_size=font_size, window_size=window_size, stride=stride)
```

Comment:
Btw, general advice 😅
Good naming is also very important, e.g. you use `ocr_flag` and `VTRSlicerOCR` which is logical since you will train a model with an OCR head. But if you think about architecture and class relation, you would see that neither the `Dataset` class nor `Slicer` doesn't know about OCR, they just return text. What happens next with the text they don't care 🤷‍♂️